mood-analysis.rb

```diff
@@ -28,5 +36,51 @@ def analyze_mood(words)
   "05/11 Yay, yay, yay! I'm having a awfuly great day."
 ]
 
-puts analyze_mood(text[0])
-puts analyze_mood(text[1])
+sad_text = [
+  "03/01 I'm having a terrible horrible no good day.",
+  "03/13 Yesterday was horrible, but today is horrible!",
+  "04/02 Sad Panda. #terribleday",
+  "04/15 Hello World, today is horrible! ",
+  "05/01 Great! Yay! Good! Yay! Happy. Happy.",
+  "05/11 Yay, yay, yay! I'm having a awfuly great day."
+]
+
+def happy_days(entries)
+  happy = 0
+  no_of_days = 0
+    entries.each do |day|
+      no_of_days += 1
+      mood = analyze_mood(day)
+      if mood == ":-)"
+        happy += 1
+      end
+      break if happy == 3
+    end
+  print "It takes #{no_of_days} days for 3 happy days to occur."
+end
```

Comment:
I would have prefered to see this method return the smallest piece of information asked for. In this case, a fixnum representing the number of days it took to hit 3 happy days, instead of a string containing that infomration. That would allow more flexibility in using this method for different situations rather than being bound to this specific use case.